tools/config-gen/parser.go

```diff
@@ -38,17 +39,17 @@ type Param struct {
 }
 
 func parseParamsConfig() ([]Param, error) {
-	yamlFile, err := os.ReadFile(*paramsFile)
+	buf, err := os.ReadFile(*paramsFile)
 	if err != nil {
 		return nil, err
 	}
 	var paramsConfig []Param
-	err = yaml.Unmarshal(yamlFile, &paramsConfig)
-	if err != nil {
+	dec := yaml.NewDecoder(bytes.NewReader(buf))
```

Comment:
yaml.Unmarshal documentation says:
```
The type of the decoded values should be compatible with the respective values in out. If one or more values cannot be decoded due to a type mismatches, decoding continues partially until the end of the YAML content, and a *yaml.TypeError is returned with details for all missed values.
```
Ref: https://pkg.go.dev/gopkg.in/yaml.v3#Unmarshal

Are we solving something different here?

Reply:
Maybe it handles type mismatches. But if you add a new field in the YAML, there is no error.